packages/yarnpkg-pnp/sources/loader/nodeUtils.ts

```diff
@@ -36,3 +37,20 @@ export function readPackage(requestPath: NativePath) {
 
   return JSON.parse(fs.readFileSync(jsonPath, `utf8`));
 }
+
+// https://github.com/nodejs/node/blob/972d9218559877f7fff4bb6086afacac8933f8d1/lib/internal/errors.js#L1450-L1478
+// Our error isn't as detailed since we don't have access to acorn to check
+// if the file contains ESM syntax
+export function ERR_REQUIRE_ESM(filename: string, parentPath: string | null = null) {
+  const basename = parentPath && path.basename(filename) ===
+    path.basename(parentPath) ? filename : path.basename(filename);
```

Comment:
I know it's taken from the Node codebase but ending the line with `===` is quite an unusual pattern for ours.

Reply:
Indeed, our linting/formatting setup should be updated to handle this.